doc/releases/changelog-dev.md

```diff
@@ -14,6 +14,9 @@
 
 * Upgrade Pauli arithmetic addition `pw1 + pw2 + 1`.
   [(#5001)](https://github.com/PennyLaneAI/pennylane/pull/5001)
+
+* Upgrade Pauli arithmetic with subtraction `w1 - w2 - 1`.
```

Comment:
It's not clear without context what `w1`, `w2` etc are, I think this would be clearer if it said something like "Subtraction between PauliWord, PauliSentence and scalars is implemented."